lib/core/src/lib/form/components/widgets/date/date.widget.ts

```diff
@@ -52,34 +55,30 @@ export class DateWidgetComponent extends WidgetComponent implements OnInit, OnDe
 
     DATE_FORMAT = 'DD-MM-YYYY';
 
-    minDate: Moment;
-    maxDate: Moment;
+    minDate: string;
+    maxDate: string;
 
     private onDestroy$ = new Subject<boolean>();
 
     constructor(public formService: FormService,
-                private dateAdapter: DateAdapter<Moment>,
-                private userPreferencesService: UserPreferencesService) {
+                private dateAdapter: DateAdapter<DateFnsAdapter>,
+                private userPreferencesService: UserPreferencesService,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig: MatDateFormats,
+                private translationService: TranslationService) {
         super(formService);
     }
 
     ngOnInit() {
         this.userPreferencesService
             .select(UserPreferenceValues.Locale)
             .pipe(takeUntil(this.onDestroy$))
-            .subscribe(locale => this.dateAdapter.setLocale(locale));
+            .subscribe(locale => this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale)));
 
-        const momentDateAdapter = this.dateAdapter as MomentDateAdapter;
-        momentDateAdapter.overrideDisplayFormat = this.field.dateDisplayFormat;
+        this.dateFormatConfig.display.dateInput = this.field.dateDisplayFormat;
 
         if (this.field) {
-            if (this.field.minValue) {
-                this.minDate = moment(this.field.minValue, this.DATE_FORMAT);
-            }
-
-            if (this.field.maxValue) {
```

Comment:
you removed the check for null/undefined, if there's no particular reason for that please restore it

Reply:
added back